java/pattern/constraint/ThingConstraint.java

```diff
@@ -445,7 +436,12 @@ private Has(@Nullable TypeVariable type, ThingVariable<?> attribute) {
 
         @Override
         public Set<BoundVariable> variables() {
-            return set(attribute);
+            Set<BoundVariable> variables = new HashSet<>();
+            variables.add(attribute);
+            if (attribute.predicate().map(p -> p.predicate.predicate().equals(TypeQLToken.Predicate.Equality.EQ)).orElse(false)) {
```

Comment:
`Predicate` class has a `variables()` method right?

Reply:
Yes. Using that directly now.